sysfs/vulnerability.go

```diff
@@ -24,33 +24,47 @@ import (
 )
 
 const (
-	notAffected = "Not Affected"
-	vulnerable  = "Vulnerable"
-	mitigation  = "Mitigation"
+	notAffected = "not affected" // based on: https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-devices-system-cpu
+	vulnerable  = "vulnerable"
+	mitigation  = "mitigation"
+)
+
+const (
+	VulnerabilityStateNotAffected = iota
+	VulnerabilityStateVulnerable
+	VulnerabilityStateMitigation
+)
+
+var (
+	VulnerabilityHumanEncoding = map[int]string{
```

Comment:
Can't find this being used anywhere?

Reply:
It's just a package variable holding the mapping between the state and a string describing the state. It's not used anywhere in this PR. It can be used for example by external consumers of this library to get a string describing the state of the vulnerability.

I've got a `node_exporter` PR waiting to be opened which is blocked on this PR. The `node_exporter` PR requires strings in order to fill labels. `node_exporter` will get a map containing vulnerabilities with encoded states and then will use this variable to map the states to more human friendly strings.

It's just an optimization I thought about. Rather than storing the exact same strings multiple times (in each Vulnerability) I thought it would make sense to encode those and store ints instead.

One could argue that I over engineered this or that it's premature optimization. I could do some benchmarks to see if it actually makes a difference, but it's a tiny change so I'm not sure if it's worth the effort. One of the maintainers could weigh in and say that simplicity is far more important at this stage (which makes total sense to me as well).

I'm open to any suggestions and feedback. Let me know what you think.

Reply:
Adding a Go doc string here would help.

```suggestion
	// VulnerabilityHumanEncoding something something
	VulnerabilityHumanEncoding = map[int]string{
```